src/blueprint/operators/protected.rs

```diff
@@ -10,6 +10,10 @@ pub fn update_protected<'a>(
     TryFold::<(&ConfigModule, &Field, &config::Type, &'a str), FieldDefinition, String>::new(
         |(config, field, type_, _), mut b_field| {
             if field.protected.is_some() || type_.protected.is_some() {
+                if type_.input {
+                    return Valid::fail("Input types can not be protected".to_owned());
```

Comment:
```suggestion
                    return Valid::fail("@protected can't be used on input types".to_owned());
```
this sounds more natural to me, let me know what you think @tusharmath